ArchiSteamFarm/IPC/Integration/ApiAuthenticationMiddleware.cs

```diff
@@ -87,18 +87,22 @@ internal sealed class ApiAuthenticationMiddleware {
 				throw new InvalidOperationException(nameof(ClearFailedAuthorizationsTimer));
 			}
 
-			string? ipcPassword = ASF.GlobalConfig != null ? ASF.GlobalConfig.IPCPassword : GlobalConfig.DefaultIPCPassword;
-
-			if (string.IsNullOrEmpty(ipcPassword)) {
-				return HttpStatusCode.OK;
-			}
-
 			IPAddress? clientIP = context.Connection.RemoteIpAddress;
 
 			if (clientIP == null) {
 				throw new InvalidOperationException(nameof(clientIP));
 			}
 
+			string? ipcPassword = ASF.GlobalConfig != null ? ASF.GlobalConfig.IPCPassword : GlobalConfig.DefaultIPCPassword;
+
+			if (string.IsNullOrEmpty(ipcPassword)) {
+				if (IPAddress.IsLoopback(clientIP) || Startup.KnownNetworks.Any(network => network.Contains(clientIP))) {
```

Comment:
This will not work if .NET maps client IPv4 address to IPv6 address space - in this case `IsIPv4MappedToIPv6` property would be `true`. It can be fixed by calling `MapToIPv4` if that property is true.

Reply:
Example (how does ASP.NET Core do it): https://github.com/dotnet/aspnetcore/blob/78ab4bd673c1040cf59400be76daf395062bc6a7/src/Middleware/HttpOverrides/src/ForwardedHeadersMiddleware.cs#L366

Reply:
Good catch, check if https://github.com/JustArchiNET/ArchiSteamFarm/commit/282828e81960aa46727bc1382cd333638e410501 solves it (it should). `IsLoopback()` apparently already handles this case according to source code and https://github.com/dotnet/corefx/pull/35463

Reply:
I also added follow-up in https://github.com/JustArchiNET/ArchiSteamFarm/pull/2372/commits/9aea006b4265618cc0b63ac46618807ed629ad93 to match ASP.NET Core logic (now we check both against IPv4 and IPv6 in known networks)

Reply:
Yes, that commit solved it.